ROADMAP.md

```diff
@@ -0,0 +1,85 @@
+# Rustls development roadmap
+
+## Future priorities
+
+In rough order of priority:
+
+* **FIPS Certification for Default Cryptographic Library**.
+  Change the default cryptographic library to something with FIPS certification.
+  rustls/rustls#1540
+
+* **Comprehensive Performance Benchmarking**.
+  Performance should be a headline feature of Rustls. We need to develop a more
+  comprehensive benchmarking system so that we can assess and improve performance
+  from multiple angles, including CPU usage, latency, and memory usage.
+
+* **Add No-Allocation / Write-Through API**.
+  Would make handshakes faster and give the caller more control over allocations.
+  RFC: rustls/rustls#1420
+
+* **Support no_std**.
+  Enables use of rustls in more memory-constrained environments.
+  RFC: rustls/rustls#1399
+
+* **OpenSSL API Compatibility Layer**.
+  Add an OpenSSL C API compatibility layer for adoption purposes.
+
+* **Support Encrypted Client Hello**.
+  Encrypted Client Hello is an upcoming standard from the TLS WG providing better
+  production for some of the data sent by a client in the initial ClientHello
+  message.
+  rustls/rustls#508
+
+* **Improve OS Trust Verifier Support**.
+  While we currently have a way to trust certificates stored in the platform trust
+  store, platform trust stores can have other ways of restricting how/when roots
+  that they expose are trusted. In order to rely on these (on Darwin and Windows)
+  we should rely on the platform verifier directly.
+
+  Given that platform verifiers may require blocking I/O, some API changes are
+  required.
+  rustls/rustls-native-certs#25
+
+* **Additional Performance Optimization**.
+  Additional performance optimization including CPU usage, latency, and memory
+  usage. The goal is to outperform OpenSSL across the board if we are not already.
+
+* **Support RFC 8879 Certificate Compression**.
+  Support for a TLS extension that substantially shrinks certificates (one of the
+  largest parts of the TLS handshake), improving handshake latency by decreasing
+  bandwidth used.
+  rustls/rustls#534
+
+* **Enforce Confidentiality / Integrity Limits**.
+  The QUIC use of TLS mandates limited usage of AEAD keys. While TLS 1.3 and 1.2
+  do not require this, the same kinds of issues can apply here, and we should
+  consider implementing limits for TLS over TCP as well.
+  rustls/rustls#755
+
+## Past priorities
+
+Delivered in 0.22:
+
+* **Enable Pluggable Cryptographic Back-ends**.
+  Allow plugging in different cryptographic back-ends.
+  rustls/rustls#1184
+
+Delivered in 0.21:
+
+* **Support IP Address Certificates**.
+  There are some popular use cases where applications want TLS certificates for
+  services that don’t have their own host name, relying on the IP address directly
+  instead. I've heard from one customer that this is common in Kubernetes deployments.
+  rustls/rustls#184
```

Comment:
editorial: I think the first person narrative here is a bit odd compared to the rest of the document: 
```suggestion
  instead. This is common in Kubernetes deployments.
  rustls/rustls#184
```